bool roll
bool pitch
bool yaw
bool sway
bool depth
bool pos_x
bool pos_y
bool pos_z
